cores/esp8266/flash_hal.cpp

```diff
@@ -28,141 +28,27 @@ extern "C" {
 #include "c_types.h"
 #include "spi_flash.h"
 }
-/*
- spi_flash_read function requires flash address to be aligned on word boundary.
- We take care of this by reading first and last words separately and memcpy
- relevant bytes into result buffer.
-
-alignment:       012301230123012301230123
-bytes requested: -------***********------
-read directly:   --------xxxxxxxx--------
-read pre:        ----aaaa----------------
-read post:       ----------------bbbb----
-alignedBegin:            ^
-alignedEnd:                      ^
-*/
 
 int32_t flash_hal_read(uint32_t addr, uint32_t size, uint8_t *dst) {
     optimistic_yield(10000);
 
-    uint32_t result = FLASH_HAL_OK;
-    uint32_t alignedBegin = (addr + 3) & (~3);
-    uint32_t alignedEnd = (addr + size) & (~3);
-    if (alignedEnd < alignedBegin) {
-        alignedEnd = alignedBegin;
-    }
-
-    if (addr < alignedBegin) {
-        uint32_t nb = alignedBegin - addr;
-        uint32_t tmp;
-        if (!ESP.flashRead(alignedBegin - 4, &tmp, 4)) {
-            DEBUGV("_spif_read(%d) addr=%x size=%x ab=%x ae=%x\r\n",
-                __LINE__, addr, size, alignedBegin, alignedEnd);
-            return FLASH_HAL_READ_ERROR;
-        }
-        memcpy(dst, ((uint8_t*) &tmp) + 4 - nb, nb);
-    }
-
-    if (alignedEnd != alignedBegin) {
-        if (!ESP.flashRead(alignedBegin, (uint32_t*) (dst + alignedBegin - addr),
-                alignedEnd - alignedBegin)) {
-            DEBUGV("_spif_read(%d) addr=%x size=%x ab=%x ae=%x\r\n",
-                __LINE__, addr, size, alignedBegin, alignedEnd);
-            return FLASH_HAL_READ_ERROR;
-        }
-    }
-
-    if (addr + size > alignedEnd) {
-        uint32_t nb = addr + size - alignedEnd;
-        uint32_t tmp;
-        if (!ESP.flashRead(alignedEnd, &tmp, 4)) {
-            DEBUGV("_spif_read(%d) addr=%x size=%x ab=%x ae=%x\r\n",
-                __LINE__, addr, size, alignedBegin, alignedEnd);
-            return FLASH_HAL_READ_ERROR;
-        }
-
-        memcpy(dst + size - nb, &tmp, nb);
+    // We use flashRead overload that handles proper alignment
+    if (ESP.flashRead(addr, dst, size)) {
+        return FLASH_HAL_OK;
+    } else {
+        return FLASH_HAL_READ_ERROR;
     }
-
-    return result;
 }
 
-/*
- Like spi_flash_read, spi_flash_write has a requirement for flash address to be
- aligned. However it also requires RAM address to be aligned as it reads data
- in 32-bit words. Flash address (mis-)alignment is handled much the same way
- as for reads, but for RAM alignment we have to copy data into a temporary
- buffer. The size of this buffer is a tradeoff between number of writes required
- and amount of stack required. This is chosen to be 512 bytes here, but might
- be adjusted in the future if there are good reasons to do so.
-*/
```

Comment:
Same here, these comments shouldn't be lost, but rather should be moved to the new code and get updated

Reply:
Same as in read